### ISSUE TITLE
Most uses of `object.mixin` should probably use `Object.assign` instead

### ISSUE THREAD

**Issue by cpcallen** [google/blockly#5914]
**Describe the bug**

There appear to be many places in the codebase where we use `object.mixin` (from `Blockly.utils.object`) when it would probably be better to use `Object.assign`.

Here's the implementation of `object.mixin`:

```JS
const mixin = function(target, source) {
  for (const x in source) {
    target[x] = source[x];
  }
};
```

Note that this will copy all enumerable properties from `source` to `target`, _including any inherited enumerable properties_.

In comparison, `Object.assign` copies only enumerable _own_ properties from the target object(s).

Since in most cases the source object is likely to be a simple string-keyed map, it is probably not normally the case that we intend or want to copy any inherited `enumerable` properties (including, for instance, any enumerable properties that an unwary developer might create on `Object.prototype`).

I therefore propose that uses of `object.mixin` be audited and, unless there is evidence that we _want_ to copy inherited properties, the call replaced with a call to `Object.assign` instead.

(I suspect this will result in 0 remaining uses of `object.mixin`, which can then be deprecated and removed.)

**Comment by derwehr** [comment]
Hi, I'd like to get started and submit a PR for this issue, if no one is already working on it

**Comment by BeksOmega** [comment]
> Hi, I'd like to get started and submit a PR for this issue, if no one is already working on it

Hello! Thank you for your interest :D Sadly at the moment we're working on a conversion to TypeScript, which may make this issue irrelevant.

Where we would really appreciate help (if you're interested) is actually in the samples repository, which adds cool new plugins and features to Blockly. Here is a [list of the help-wanted issues](https://github.com/google/blockly-samples/labels/help%20wanted).

If there's anything we can do to help you get started, don't hesitate to ask!

**Comment by cpcallen** [comment]
Closed by #6138.
